core/src/main/java/org/kiwix/kiwixmobile/core/main/NightModeViewPainter.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.graphics.ColorMatrixColorFilter
+import android.graphics.Paint
+import android.view.View
+import android.widget.VideoView
+import org.kiwix.kiwixmobile.core.NightModeConfig
+import javax.inject.Inject
+
+/**
+ * NightModeViewPainter class is used to apply respective filters to the views
+ * depending whether the app is in dark mode or not
+ * Created by yashk2000 on 24/03/2020.
+ */
+
+class NightModeViewPainter @Inject constructor(
+  private val nightModeConfig: NightModeConfig
+) {
+
+  private val invertedPaint =
+    Paint().apply { colorFilter = ColorMatrixColorFilter(KiwixWebView.NIGHT_MODE_COLORS) }
+
+  fun <T : View> update(
```

Comment:
you can give `vararg additionalViews: View` a default value of `emptyArray()` and then annotate this method with `@JvmOverloads`

This will create 3 different versions of this method you can call from java to ease the interop.

Reply:
@macgills could you please tell why do we need to create 3 different versions of the function? We are just calling the function from 2 places.

Reply:
It is to get compatability in java with default arguments.

in kotlin the simple case is you want to paint 1 view `painter.update(myView)`, because of default arguments you don't need to supply anything else

in java - because it sucks - you have to write `painter.update(myView, { true }, emptyArray)`, the emptyArray might not be necessary because of how varargs works but you still have to say more than you want

`@JVMOverloads` will create methods with different numbers of args with the default arguments filled in, you can read more [here](https://android.jlelse.eu/demystifying-the-jvmoverloads-in-kotlin-10dd098e6f72)

Reply:
`@JVMOverloads` is a really cool annotation. Thanks for the blog post :)